### PR TITLE
Document how to make custom min/max for Array

### PR DESCRIPTION
**Pull request by AThousandShips**
From godotengine/godot-proposals#6609

(unfortunately not available for 3.x as `reduce` isn't available there)

### CODE REVIEW

File: doc/classes/Array.xml [hunk 1]
```diff
@@ -441,12 +441,32 @@
 			<return type="Variant" />
 			<description>
 				Returns the maximum value contained in the array if all elements are of comparable types. If the elements can't be compared, [code]null[/code] is returned.
+				To find the minimum value using a custom comparator, use [method reduce] like this (this returns the first maximum value if multiple values would be considered the maximum, as in the example):
+				[codeblock]
+				func _ready():
+				    var arr = [Vector2(0, 1), Vector2(2, 0), Vector2(1, 1), Vector2(1, 0), Vector2(0, 2)]
+				    # The function cmp(a, b) should return true if a &lt; b, here we compare the lengths.
```
Comment (MewPurPur) [inline doc/classes/Array.xml:448]: May I suggest "The function cmp(a, b) returns true if a is longer than b"? Or at least replace "should return" with "returns" to sound more final. Likewise below.

Comment (AThousandShips) [reply doc/classes/Array.xml:448]: The "should" is to indicate how the function should behave, indicating the requirement on it

Comment (MewPurPur) [reply doc/classes/Array.xml]: I don't think stating requirements makes sense here, since you implemented it below and it's just an example.

Comment (MewPurPur) [reply doc/classes/Array.xml]: You missed an "if" in the new wording.

Comment (AThousandShips) [reply doc/classes/Array.xml]: Sure, will correct

**Comment by AThousandShips** [comment]
True, added it to illustrate the use with callables but will make it simpler, and will cross reference with details on minimum case

File: doc/classes/Array.xml [hunk 2]
```diff
@@ -441,12 +441,23 @@
 			<return type="Variant" />
 			<description>
 				Returns the maximum value contained in the array if all elements are of comparable types. If the elements can't be compared, [code]null[/code] is returned.
+				To find the maximum value using a custom comparator, use [method reduce] like this (this returns the first maximum value if multiple values would be considered the maximum, as in the example):
```
Comment (YuriSizov) [inline doc/classes/Array.xml:444]:
```suggestion
				To find the maximum value using a custom comparator, you can use [method reduce]. In this example every array element is checked and the first maximum value is returned:
```

File: doc/classes/Array.xml [hunk 3]
```diff
@@ -441,12 +441,23 @@
...
+				[codeblock]
+				func _ready():
+				    var arr = [Vector2(0, 1), Vector2(2, 0), Vector2(1, 1), Vector2(1, 0), Vector2(0, 2)]
+				    # In this example we compare the lengths.
+				    print(arr.reduce(func(max, val): return val if cmp_len(val, max) else max))
+
+				func cmp_len(a, b):
```
Comment (YuriSizov) [inline doc/classes/Array.xml:451]:
```suggestion
				    print(arr.reduce(func(max, val): return val if is_length_greater(val, max) else max))

				func is_length_greater(a, b):
```

Prefer full names for methods. We can also make it more explicit here, by naming it after what it does. (It also helps with reading the ternary expression).

**Review by YuriSizov — APPROVED** [review]

**Comment by YuriSizov** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank you!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.0.3.

**Pull request merged**